### PR TITLE
Add acl.enable_key_list_policy to agent config docs.

### PR DESCRIPTION
**Pull request by kaitlincarter-hc**

### CODE REVIEW

File: website/source/docs/agent/options.html.md [hunk 1]
```diff
@@ -551,7 +551,9 @@ default will automatically work with some tooling.
      ACLs are a blacklist: any operation not specifically prohibited is allowed. In "deny" mode, ACLs are
      a whitelist: any operation not specifically allowed is blocked. *Note*: this will not take effect until
      you've enabled ACLs.
-
+     
+     * <a name="acl_enable_key_list"></a><a href="#acl_enable_key_list">`enable_key_list`</a> - Either "true" or "false", defaults to "false". When enabled, it allows access to recursively list Consul KV enteries and keys. It also enables more fine grained policies. This option is only available in Consul 1.0 and newer.
```
Comment (banks) [inline website/source/docs/agent/options.html.md]: Observation: we are quite inconsistent with the way we describe the types and possible values in this page. This follows the precedent from others around it so I think it's fine, but typically we don't explicitly say that boolean fields are `Either "true" or "false"` and I guess assume that booleans are implied and understood by `enable_*` or `disable_*` prefix.

tl;dr: this is fine, just wondering for the future if we can make our descriptions more consistent.

One other nit: it's not totally consistent, but we tend to try to keep lines in the source to a max width of 80-100 ish chars rather than letting them wordwrap. Many devs don't enable wordwrap in editors and even on a large screen have many narrow panes so it makes them harder to work with. These docs are pretty inconcistent but for the most part they do break.

Personally I use an editor extension that reflows the text in current paragraph with a keyboard shortcut as it's painful trying to manually wrap prose as you edit.

Not necessary for landing just an observation - I sometimes end up reformatting our source when lines are long around a part I'm editing!

Comment (banks) [reply website/source/docs/agent/options.html.md]: One other comment I didn't even know about this flag? Does this mean that by default if ACLs are not permissive you can't do a recurse lookup even if you have ACL read on every key it would touch? That seems confusing, but if it's not the case then we might need to be a little more specific with this wording. Maybe @mkeeler has more of an idea?

Comment (mkeeler) [reply website/source/docs/agent/options.html.md]: So the way the key list permission works is that iff `acl.enable_key_list` is `true` then we ensure kv listing operations on a prefix have key list on that prefix. When disabled we allow the list to work but then filter the result set based on other key acl rules.

Comment (mkeeler) [reply website/source/docs/agent/options.html.md]: With a policy like:

```
key_prefix "foo/" {
   policy = "read"
}
```

With `enable_key_list` set to `false` then this alone allow discovering and reading any key under `foo/`. Once enabled this will allow reading keys if you already know what they are but not listing out the entire prefix unless the policy is:

```
key_prefix "foo/" {
   policy = "list"
}
```

One other note is that `list` permissions implicitly also grant `read`, so the total precedence of all policy dispositions is:

1. Deny
2. Write
3. List
4. Read

Comment (mkeeler) [inline website/source/docs/agent/options.html.md]: The following is more accurate for what actually happens when you enable this:

> When enabled, the `list` permission will be required on the prefix being recursively read in the KV store. Regardless of being enabled, the full set of KV entries under the prefix will be filtered to remove any entries that the request's ACL token doesn't grant at least read permissions.

Comment (kaitlincarter-hc) [reply website/source/docs/agent/options.html.md]: Paul - I noticed the language/style inconsistencies in that doc too. I think this will be a good page to focus on while creating the style guide. Also noted, I will setup text wrapping :).
